File: TikraqApp/src/app/model/Traduccion.ts
export class Traduccion {
  id: number;
  textoOriginal: string;
  textoTraducido: string;
  fechaTraduccion: Date;
  idiomaOriginal: string;
  idiomaTraducir: string;
  usuariosUsuarioid: { id: number };

  constructor(id: number, textoOriginal: string, textoTraducido: string, fechaTraduccion: Date, idiomaOriginal: string, idiomaTraducir: string, usuariosUsuarioid: { id: number }) {
    this.id = id;
    this.textoOriginal = textoOriginal;
    this.textoTraducido = textoTraducido;
    this.fechaTraduccion = fechaTraduccion;
    this.idiomaOriginal = idiomaOriginal;
    this.idiomaTraducir = idiomaTraducir;
    this.usuariosUsuarioid=usuariosUsuarioid;
  }
}

File: Tikraq/src/main/java/com/example/tikraq/entities/Traduccion.java
package com.example.tikraq.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
@Entity
@Table(name = "traducciones")
public class Traduccion {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String textoOriginal;
    private String textoTraducido;
    private LocalDate fechaTraduccion;
    private String idiomaOriginal;
    private String idiomaTraducir;
    @ManyToOne
    @JoinColumn(name = "usuarios_usuarioid")
    private Usuario usuariosUsuarioid;

}